### PR TITLE
Compute: support specialized gallery image version

### PR DESCRIPTION
**Pull request by xccc-msft**
Fix: [Azure/azure-sdk-for-java#11912](https://github.com/Azure/azure-sdk-for-java/issues/11912)

### CODE REVIEW

File: azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/VirtualMachineImpl.java [hunk 1]
```diff
@@ -720,6 +722,16 @@ public VirtualMachineImpl withSpecializedOSDisk(Disk disk, OperatingSystemTypes
         return this;
     }
 
+    @Override
+    public VirtualMachineImpl withSpecializedGalleryImageVersion(String galleryImageVersionId) {
+        ImageReference imageReferenceInner = new ImageReference();
+        imageReferenceInner.withId(galleryImageVersionId);
+        this.inner().storageProfile().osDisk().withCreateOption(DiskCreateOptionTypes.FROM_IMAGE);
```
Comment (ChenTanyi) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/VirtualMachineImpl.java:729]: This makes `isOSDiskFromImage` true, so no need another bool.

Comment (xccc-msft) [reply azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/implementation/VirtualMachineImpl.java:729]: Only if it is `specialized`, we need to set OS profile as null. If it is `generalized`, keep as it is.

File: azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java [hunk 2]
```diff
@@ -913,6 +904,14 @@
...
              */
             WithLinuxRootPasswordOrPublicKeyManagedOrUnmanaged withRootUsername(String rootUserName);
+
+            /**
+             * Specifies to remove the information of OS profile for the Linux virtual machine.
+             * It is required when creating from a specialized image.
+             *
+             * @return the next stage of the definition
+             */
+            WithLinuxCreateManagedOrUnmanaged withoutRootUsernameAndPassword();
```
Comment (weidongxu-microsoft) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java:914]: Suggestion: Maybe same method like `withNoCredential` for both windows and linux flow?

Comment (xccc-msft) [reply azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java:914]: In vm/vmss, it should be OK as we already have existing methods like:
`withoutPrimaryPublicIPAddress`: the VM should not have a public IP address.
`withoutAutoUpdate`: disables automatic updates.

**Comment by weidongxu-microsoft** [comment]
One question: why not have e.g.
```
WithWindowsCreateManagedOrUnmanaged withWindowsSpecializedGalleryImageVersion(String galleryImageVersionId);
```
which would skip the username/password stage?

**Comment by xccc-msft** [comment]
> One question: why not have e.g.
> 
> ```
> WithWindowsCreateManagedOrUnmanaged withWindowsSpecializedGalleryImageVersion(String galleryImageVersionId);
> ```
> 
> which would skip the username/password stage?

It should also apply to other cases when passing specialized image(id or url)/image version. So might be better having `without*` as per my opinion.

**Comment by weidongxu-microsoft** [comment]
> > One question: why not have e.g.
> > ```
> > WithWindowsCreateManagedOrUnmanaged withWindowsSpecializedGalleryImageVersion(String galleryImageVersionId);
> > ```
> > 
> > 
> > which would skip the username/password stage?
> 
> It should also apply to other cases when passing specialized image(id or url)/image version. So might be better having `without*` as per my opinion.

This appears to be the only method involving `Gallery`. Do we aware any other cases that having specialized image?

And we already have a `withSpecializedOSDisk` method. Please check whether this is for specialized OS.

**Comment by xccc-msft** [comment]
> > > One question: why not have e.g.
> > > ```
> > > WithWindowsCreateManagedOrUnmanaged withWindowsSpecializedGalleryImageVersion(String galleryImageVersionId);
> > > ```
> > > 
> > > 
> > > which would skip the username/password stage?
> > 
> > 
> > It should also apply to other cases when passing specialized image(id or url)/image version. So might be better having `without*` as per my opinion.
> 
> This appears to be the only method involving `Gallery`. Do we aware any other cases that having specialized image?
> 
> And we already have a `withSpecializedOSDisk` method. Please check whether this is for specialized OS.

For methods like `withLinuxCustomImage(String customImageId)`, it may also have specialized. For `withSpecializedOSDisk`, it should be fine as the next phase doesn't require username/password input.

**Comment by weidongxu-microsoft** [comment]
For `withLinuxCustomImage`, can you confirm that it supports specialized image? Any doc or tests?

For `withSpecializedOSDisk`, what I mean is that it appears already THE flow for specialized OS (as a existing OS disk -- apparently specialized, instead of a specialized image).

**Comment by xccc-msft** [comment]
> For `withLinuxCustomImage`, can you confirm that it supports specialized image? Any doc or tests?
> 
> For `withSpecializedOSDisk`, what I mean is that it appears already THE flow for specialized OS (as a existing OS disk -- apparently specialized, instead of a specialized image).

The specialized/generialized is set on image level. Under the image, you could have different versions like `0.0.1`/`0.0.2`, which is the image version reported by our user.
The OS profile would only be passed when importing image. When importing OS disk, the OS profile will be set as null.

**Pull request merged**